### ISSUE TITLE
VPN Configuration

### ISSUE THREAD

**Issue by dvddarias** [ukanth/afwall#713]
Is there a way to configure the following behavior:

- if the VPN is connected via WIFI or LAN then allow all apps to go through it, but if the VPN is connected using Mobile Data then only allow specific apps???

Thanks in advance.

**Comment by benzntech** [comment]
Searching for the same Configuration..

**Comment by walrus543** [comment]
Nope. Actually this has been discussed on XDA thread and no solution has been found (yet).

**Comment by exec228** [comment]
Hello!
Any updates?
Let's consider VPN is "always on".
1) I want maps, offline dictionaries to update only when VPN is on Wi-Fi. Not Mobile data, to save the plan.
2) I want email, banking, weather, remote desktop to connect either when VPN is on Wi-Fi, or Mobile data.
3) I want, just for example, to A-GPS/Ephemeris to update when VPN is on Mobile data only, not Wi-Fi.
Is it possible? May be it's already implemented?

**Comment by benzntech** [comment]
@ukanth : Please review the the open ticket . This ticket is not addressed for years. Please dont keeps us waiting.

**Comment by ukanth** [comment]
This can be addressed using Profiles ?

**Comment by protist** [comment]
> This can be addressed using Profiles ?

That looks promising, but is there a way to automatically switch profile based on whether the user is connected to WiFi or mobile data? (I had a look at the broadcast receivers via [Automate](https://llamalab.com/automate/), but couldn't find a way to trigger this externally either.)

I was actually confused by the existing Wifi/mobile icons. From my testing, it seems like that these settings are totally ignored if you are using a VPN? I would have expected them to have some effect still. FWIW I know it's possible for apps (e.g. [AntennaPod](https://github.com/AntennaPod/AntennaPod)) to determine if you are connected to Wifi or mobile, even with a VPN active. Ideally I would expect AFWall+ also to make this explicit distinction.
